### PR TITLE
boards: stm32: Fix boards names in yaml and/or doc files

### LINKED ISSUE #24927: stm32: Fix docs boards for doc generation
**Describe the bug**
For automatic board generation, value of name yaml property should match name in board file .rst:
https://github.com/zephyrproject-rtos/zephyr/blob/master/boards/arm/96b_neonkey/doc/index.rst#96boards-neonkey
should match: https://github.com/zephyrproject-rtos/zephyr/blob/e1dddf7befa7309bd2afc567b2e00d2e7362f7c4/boards/arm/96b_neonkey/96b_neonkey.yaml#L2

Check this for whole set of STM32 boards.
Additionally, check board images are fine (right names, ..)

### PR DESCRIPTION
**Pull request by erwango**
For atuo doc generation purpose, get name value of boards' yaml files
in sync with name provided as board name in .rst file

Fixes #24927

Signed-off-by: Erwan Gouriou <erwan.gouriou@linaro.org>

### CODE REVIEW

File: boards/arm/stm32g0316_disco/doc/index.rst [hunk 1]
```diff
@@ -1,6 +1,6 @@
 .. _stm32g0316_disco_board:
 
-STM32G0316-DISCO
+ST STM32G0316 Discovery
 ################
```
Comment (galak) [inline boards/arm/stm32g0316_disco/doc/index.rst:4]: Need to fix the hash's here

Comment (erwango) [reply boards/arm/stm32g0316_disco/doc/index.rst]: The change was in the editor buffer. Fixed, txs!

**Comment by erwango** [comment]
@nashif, fixed and updated board names.

A question on <board>/ folders with 2 targets (for instance dual core stm32h747i_disco):
There is one yaml file per target, but only one overall doc.
Is that ok to have same name yaml property for both targets ?

**Review by galak — APPROVED** [review]

**Review by nashif — APPROVED** [review]

**Comment by nashif** [comment]
> Is that ok to have same name yaml property for both targets ?

I guess yes, need to look how we can improve this in the future, for now however it should work.

**Comment by erwango** [comment]
> I guess yes, need to look how we can improve this in the future, for now however it should work.

Ok, I did the opposite. Let me update.

**Comment by galak** [comment]
Since this touches some many ST boards CI will timeout, looking at looks everything is building so going to merge by hand.

**Pull request merged**